vehicle/src/Vehicle/Compile/Type/Monad/Class.hs

```diff
@@ -109,49 +111,42 @@ class (MonadCompile m, MonadFreeContext builtin m, NormalisableBuiltin builtin,
   modifyTypeCheckerState :: (TypeCheckerState builtin -> TypeCheckerState builtin) -> m ()
   getFreshName :: Type builtin -> m Name
   clearFreshNames :: Proxy builtin -> m ()
-  getInstanceCandidates :: m (InstanceDatabase builtin)
```

Comment:
Removing this is really nice, but could we make it a derived method instead in the `Operations` section of this file? That would minimise the changes needed upstream!